pymmmc/mmm/preprocessing.py

```diff
@@ -0,0 +1,58 @@
+from typing import Callable
+
+import pandas as pd
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import MaxAbsScaler, MinMaxScaler, StandardScaler
+
+__all__ = [
+    "preprocessing_method",
+    "MixMaxScaleTarget",
+    "MaxAbsScaleChannels",
+    "StandardizeControls",
+]
+
+
+def preprocessing_method(method: Callable) -> Callable:
+    if not hasattr(method, "_tags"):
+        method._tags = {}
+    method._tags["preprocessing"] = True
+    return method
+
+
+class MixMaxScaleTarget:
+    @preprocessing_method
+    def min_max_scale_target_data(self, data_df: pd.DataFrame) -> pd.DataFrame:
+        target_vector = data_df[self.target_column].to_numpy().reshape(-1, 1)
+        transformers = [("scaler", MinMaxScaler())]
+        pipeline = Pipeline(steps=transformers)
+        self.target_transformer: Pipeline = pipeline.fit(X=target_vector)
+        data_df[self.target_column] = self.target_transformer.transform(
+            X=target_vector
+        ).flatten()
+        return data_df
+
+
+class MaxAbsScaleChannels:
+    @preprocessing_method
+    def max_abs_scale_channel_data(self, data_df: pd.DataFrame) -> pd.DataFrame:
+        channel_data: pd.DataFrame = data_df[self.channel_columns]
+        transformers = [("scaler", MaxAbsScaler())]
+        pipeline: Pipeline = Pipeline(steps=transformers)
+        self.channel_transformer: Pipeline = pipeline.fit(X=channel_data.to_numpy())
+        data_df[self.channel_columns] = self.channel_transformer.transform(
+            channel_data.to_numpy()
+        )
+        return data_df
+
+
+class StandardizeControls:
```

Comment:
At the moment this works for continuous controls and ordinal numeric encoded right? Maybe we want to specify this in the name as will probably want to add a default one-hot-encoding for categorical variables?

Reply:
I’ll create an issue to keep track of that